source/agora/utils/PrettyPrinter.d

```diff
@@ -70,22 +70,29 @@ private struct AmountFmt
 {
     private Amount value;
 
-    public void toString (scope void delegate(scope const(char)[]) @safe sink) @safe
+    public void toString (scope void delegate(scope const(char)[]) @safe sink) @safe nothrow
     {
-        formattedWrite(sink, "%,d", this.value.integral());
-        if (auto dec = this.value.decimal())
+        try
         {
-            sink(".");
-            size_t mask = 1_000_000;
-            while (dec)
+            formattedWrite(sink, "%,d", this.value.integral());
```

Comment:
This should be `nothrow` if the `sink` is `nothrow`, no ?

Reply:
Discussed IRL. `formattedWrite` is not nothrow because it uses Exceptions instead of Errors.